### PR TITLE
Implemented vulkan backend ops for cross entropy loss  and cross_entropy_loss_back

### PR DESCRIPTION
**Pull request by PranavUttarkar**
## Overview

Added vulkan support for GGML_OP_CROSS_ENTROPY_LOSS and `GGML_OP_CROSS_ENTROPY_LOSS_BACK (both are in one PR because they share the same setup except for the last pass and how the result is written where  CEL_back  has the full tensor with the same shape as logits, no extra kernel and scalar grad.

These  used to be unsupported on vulkan and fell back to cpu. The forward path follows the cuda/SYCL one workgroup per row, numerically stable log-softmax over classes, then a sum_rows reduce into the scalar loss pattern. The backward path is a single kernel that reuses the same max + softmax setup and writes (softmax(logits) - labels) * grad / nrows

supports_op right now is limited to contiguous F32 tensors with matching logits/labels shapes (forward output must be a scalar) A `wg512` pipeline variant is selected when `nclasses > 1024`, same threshold as `soft_max`.

Part of https://github.com/ggml-org/llama.cpp/issues/14909 

## Additional information

Modeled on the CUDA and SYCL cross-entropy paths, with some reuse from vulkan soft_max/sum_rows (row dispatch via ggml_vk_nrows_elements, prealloc_x temp buffer, existing pipeline_sum_rows_f32). Checked against the CPU reference in ggml_compute_forward_cross_entropy_loss_f32 and ggml_compute_forward_cross_entropy_loss_back_f32.

Also the shaders use a fixed BLOCK_SIZE workgroup reduce in shared memory, not nclasses-sized shared memory, so large-vocab cases like {30000,1,1,1} work without a separate large-class path.

Test results (AMD Radeon 780M, Windows):

```
test-backend-ops.exe test    -b Vulkan0 -o CROSS_ENTROPY_LOSS       2/2 passed
test-backend-ops.exe test    -b Vulkan0 -o CROSS_ENTROPY_LOSS_BACK  2/2 passed
test-backend-ops.exe support -b Vulkan0 -o CROSS_ENTROPY_LOSS       SUPPORTED
test-backend-ops.exe support -b Vulkan0 -o CROSS_ENTROPY_LOSS_BACK  SUPPORTED
test-backend-ops.exe grad    -b Vulkan0 -o CROSS_ENTROPY_LOSS       OK ({10,5,4,3}; {30000,...} skipped as large)
```

Docs: `docs/ops/Vulkan.csv` and `docs/ops.md` updated for both ops on Vulkan.

## Requirements

- I have read and agree with the [contributing guidelines](https://github.com/ggml-org/llama.cpp/blob/master/CONTRIBUTING.md)
- AI usage disclosure: 

YES: AI was used in the beginning to understand for research and navigation to find similar implementation. Code was all handwritten and then AI was used to review it a few times and look for edge cases and helped find some oversights/gaps which I then adressed.

### CODE REVIEW

**Comment by jeffbolznv** [comment]
I'm not familiar with the cross-entropy ops, but the changes here generally look fine to me. Generally I'd prefer to handle noncontiguous and unaligned tensors when we add new ops, but I'm not sure whether it's all that important for these ops.

**Review by 0cc4m — APPROVED** [review]

**Comment by 0cc4m** [comment]
I think it's okay like this for now.

**Review by jeffbolznv — APPROVED** [review]
